drivers/led/leds_gpio.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2018 Manivannan Sadhasivam
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief LEDs driver for GPIOs
+ *
+ * This driver controls LED devices attached to GPIO pins which
+ * are declared under leds node in board devicetree. The driver
+ * takes care of configuring the pin to output state before using it.
+ *
+ * TODO: Implement Software Blinking
+ */
+
+#include <board.h>
+#include <device.h>
+#include <gpio.h>
+#include <led.h>
+#include <misc/util.h>
+#include <zephyr.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_LED_LEVEL
+#include <logging/sys_log.h>
+
+#include "led_context.h"
+
+enum led_gpio_modes {
+	LED_OFF,
+	LED_ON,
+};
+
+struct led_gpio_cfg {
+	char *gpio_port;
+	u8_t gpio_pin;
+};
+
+struct led_gpio_data {
+	struct device *gpio;
+	struct led_data dev_data;
+};
+
+static inline int led_gpio_on(struct device *dev, u32_t led)
+{
+	const struct led_gpio_cfg *cfg = dev->config->config_info;
+	struct led_gpio_data *data = dev->driver_data;
+
+	/* Make sure that the LED is valid */
+	if (led != cfg->gpio_pin)
+		return -EINVAL;
+
+	gpio_pin_write(data->gpio, led, LED_ON);
```

Comment:
Many NXP boards invert the polarity, meaning you have to write a zero to the gpio to turn on the LED.

Reply:
Okay. Then the best way to address is to have a `invert-polarity` property on the `leds` node. What is your opinion?

Reply:
@MaureenHelm Just a ping over my query.

Reply:
We've already got information about the polarity in dts, can you use that?
https://github.com/zephyrproject-rtos/zephyr/blob/c1f54cc5fdec8e5180431aae151e5f4299944367/boards/arm/frdm_k64f/frdm_k64f.dts#L50-L51

Reply:
Hmm, makes sense. Thanks for the pointer!